src/formatDate.js

```diff
@@ -50,7 +50,46 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateData = date.split(fromFormat[3]);
+
+  const result = [];
+  let year = '';
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dateData[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = dateData[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = dateData[i];
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
```

Comment:
Same as above, replacing 'if' statements with a 'switch' statement would improve readability and performance. Also consider using a map or object structure for 'year', 'month', and 'day' variables.